firedrake/mesh.py

```diff
@@ -1575,10 +1568,7 @@ def __init__(self, swarm, parentmesh, name, reorder, use_cell_dm_marking, origin
 
         # Mark OP2 entities and derive the resulting Swarm numbering
         with PETSc.Log.Event("Mesh: numbering"):
-            if use_cell_dm_marking:
-                dmcommon.mark_entity_classes_using_cell_dm(self.topology_dm)
-            else:
-                dmcommon.mark_entity_classes(self.topology_dm)
+            dmcommon.mark_entity_classes(self.topology_dm)
```

Comment:
So, was the pointSF set up correctly before your changes which construct it using the input ordering SF?

Reply:
IIRC it was failing tests (though I see you've also modified the tests...)

Reply:
The pointSF you had was correct.

I was actually careless here. I thought `owned` label was not necessary for vom, so we could just use `mark_entity_classes`, which labels `ghost` entities correctly. But it was wrong, so I put back `mark_entity_using_cell_dm` (I rewrote, so please have a look). I also reverted the change that I made in the tests.

Initially, we decided to have vom inherit pyop2 labels from the parent as it was natural, but if I think about how `parloop` works in two phases (`core` loop + `owned` loop, https://github.com/OP2/PyOP2/blob/d017d594e0bda694a71c4180fbfeae1cba473e93/pyop2/parloop.py#L216), inheriting the labels actually seems necessary, and we should always do so.
`input_ordering` currently does not, so I just kept it as is for now.

Reply:
`input_ordering` also has no halos so everything should be `core`

Reply:
Do you recall why the pointSF of the VOM does not correctly label `owned` cells?

Reply:
Given a non-ghost cell `c`, if any point in the closure of `c` is ghost, we label all points in the closure as `owned` (if not ghost). That is why there is no `owned` cells (vertices) in VOMs, at least in this definition.

Reply:
That is not how we define `owned`. The idea I think is that when we do `Parloop`, we can kick off iteration over `core` cells without waiting for the halo exchange to complete, while, for iteration over `owned` cells, we have to make sure that the halo data are up-to-date.
However, as far as I understand, this way of handling the two-phase computation (`core` loop and `owned` loop) by statically marking `core` and `owned` entities does not directly work if multiple meshes are involved in the `Parloop`, and we probably need a further discussion when we do, e.g., couplings.
VOMs can be examples where multiple meshes are involved, but, if a VOM and its parent are the only ones involved, thanks to the nature of VOM, we can use existing infrastructure by simply letting VOM inherit the pyops labels from the parent; when we do interpolation onto a VOM, we need to make sure that all parent function data are up-to-date when computation is performed in `Parloop`, and inheriting `owned` label I think allows this.